src/sqlfluff/rules/L007.py

```diff
@@ -58,51 +68,103 @@ def _eval(self, context: RuleContext) -> LintResult:
-        # The parent stack tells us whether we're in an expression or not.
-        if context.parent_stack and context.parent_stack[-1].is_type("expression"):
-            if context.segment.is_code:
-                # This is code, what kind?
-                if context.segment.is_type("binary_operator", "comparison_operator"):
-                    # If it's an operator, then check if in "before" mode
-                    if self.operator_new_lines == "before":  # type: ignore
-                        # If we're in "before" mode, then check if newline since last
-                        # code
-                        for s in memory["since_code"]:
-                            if s.name == "newline":
-                                # Had a newline - so mark this operator as a fail
-                                anchor = context.segment
-                                # TODO: Work out a nice fix for this failure.
-                elif memory["last_code"] and memory["last_code"].is_type(
-                    "binary_operator", "comparison_operator"
-                ):
-                    # It's not an operator, but the last code was.
-                    if self.operator_new_lines != "before":  # type: ignore
-                        # If in "after" mode, then check to see
-                        # there is a newline between us and the last operator.
-                        for s in memory["since_code"]:
-                            if s.name == "newline":
-                                # Had a newline - so mark last operator as a fail
-                                anchor = memory["last_code"]
-                                # TODO: Work out a nice fix for this failure.
-                # Prepare memory for later
-                memory["last_code"] = context.segment
-                memory["since_code"] = []
-            else:
-                # This isn't a code segment...
-                # Prepare memory for later
-                memory["since_code"].append(context.segment)
-        else:
-            # Reset the memory if we're not in an expression
-            memory = {"last_code": None, "since_code": []}
-
-        # Anchor is our signal as to whether there's a problem
-        if anchor:
-            return LintResult(anchor=anchor, memory=memory, description=description)
-        else:
-            return LintResult(memory=memory)
+        relevent_types = ["binary_operator", "comparison_operator"]
+        segment = context.functional.segment
+        # bring var to this scope so as to only have one type ignore
+        operator_new_lines: str = self.operator_new_lines  # type: ignore
+        expr = segment.children()
+        operator_segments = segment.children(sp.is_type(*relevent_types))
+        results: List[LintResult] = []
+        # If len(operator_segments) == 0 this will essentially not run
+        for operator in operator_segments:
+            start = expr.reversed().first(FirstCodeAfterOperator(operator))
```

Comment:
Nitpick: Rather than `expr.reversed().first()`, could you use `expr.last()`?

Reply:
Is that not also confusing the `.last(FirstCodeAfterOperator(operator))`
The last First is a bit hard to get right ?